hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenAttributesValidator.java

```diff
@@ -143,25 +143,25 @@ public void validateTokenKeys(
         if (hasAdminKey && !isKeyRemoval(adminKey)) {
             validateTrue(isValid(adminKey), INVALID_ADMIN_KEY);
         }
-        if (hasKycKey) {
+        if (hasKycKey && !isKeyRemoval(kycKey)) {
```

Comment:
Why are we not checking the `NO_VALIDATION` condition here ?

Reply:
Because method `validateTokenKeys()` is also used in `TokenCreateValidator` as well. Instead, in `TokenUpdateValidator` we are checking for other than `NO_VALIDATION` enum value and then perform key validations